inference-engine/src/plugin_api/cpp_interfaces/interface/ie_iplugin_internal.hpp

```diff
@@ -214,9 +214,10 @@ class IInferencePlugin : public details::IRelease,
 
     /**
      * @brief      Provides a default remote context instance if supported by a plugin
+     * @param[in]  params  The map of parameters
      * @return     The default context.
      */
-    virtual RemoteContext::Ptr GetDefaultContext() = 0;
+    virtual RemoteContext::Ptr GetDefaultContext(const ParamMap& params) = 0;
```

Comment:
Can you please explain what this argument means, and which `params` are expected here?

Reply:
`KEY_DEVICE_ID` can be expected in case if from public API we call:
```c++
InferenceEngine::Core core;
auto context_GPU0 = core.GetDefaultContext("GPU.0");
auto context_GPU1 = core.GetDefaultContext("GPU.1");
```
where `context_GPU0` and `context_GPU1` can be different.

Reply:
hm. why `CreateContext()` cannot be used for such cases?

Reply:
I think `CreateContext()` is supposed to wrap custom shared context, not to create a new one.

Reply:
IMHO, looking to the signature of `CreateContext()` it seems like not necessary to limit it's functionality to only 'wrap custom shared context'. At least onednn plugin utilizes this method internally to implement `DEVICE_ID` support.

Reply:
Anyway, we have 2 signatures and:
```c++
InferenceEngine::Core core;
auto context_GPU0 = core.GetDefaultContext("GPU.0");
auto context_GPU1 = core.GetDefaultContext("GPU.1");
```
A valid case, so we need to pass device ID from GetDefaultContext to internal method with the same name.

Reply:
One more question: how `KEY_DEVICE_ID` param key should consort with `DEVICE_ID` configuration?

Reply:
it's the same things.

Reply:
hm, why the:
```
InferenceEngine::Core core;
auto context_GPU0 = core.CreateContext("GPU.0");
auto context_GPU1 = core.CreateContext("GPU.1");
```
is not the valid case then?

Reply:
```c++
InferenceEngine::Core core;
auto context_GPU0 = core.CreateContext("GPU.0", {});
auto context_GPU1 = core.CreateContext("GPU.1", {});
```
depends on plugin:
1. Your plugin threats this as default context.
2. GPU plugin threats this as invalid case because not all the required parameters are passed to create shared context. For explicity, `GetDefaultContext` was introduced by @mletavin 

I think 2 different methods were added for better explicity for users to identify their intentions.